### ISSUE TITLE
provider doesn't handle large zonefile results

### ISSUE THREAD

**Issue by jaxxstorm** [terraform-providers/terraform-provider-dns#16]
### Terraform Version
v0.10.4

### Affected Resource(s)
- dns_a_record_set
- I expect it affects all

### Terraform Configuration Files
```hcl
resource "dns_a_record_set" "ec2_instance" {
  zone      = "${var.domain}."
  name      = "${var.instance_prefix}-${count.index+1}"
  addresses = ["${aws_instance.ec2_instance.*.private_ip}"]
  ttl       = 300
}
```

### Debug Output
```
Error refreshing state: 1 error(s) occurred:

* module.build_server.dns_a_record_set.ec2_instance: 1 error(s) occurred:

* module.build_server.dns_a_record_set.ec2_instance: dns_a_record_set.ec2_instance: Error querying DNS record: dns: failed to unpack truncated message
```

### Expected Behavior
The A record (which has already been created) should have been returned

### Actual Behavior
Above error

### Steps to Reproduce
1. have a bunch a bunch of large zone files that exceed UDP responses
2. add a record
3. result

### Important Factoids
We have a number of large BIND dns zone files

I suspect this is similar to https://github.com/miekg/dns/issues/423 in that it's because of the failure to use TCP.

**Comment by miekg** [comment]
If you implement DNS; you need to implement the entire spec. Either retry with TCP or set an EDNS0 option to allow larger replies.

**Comment by jaxxstorm** [comment]
@miekg any tips on how to do this?

**Comment by miekg** [comment]
If you have a dns.Client you can set udpsize https://godoc.org/github.com/miekg/dns#Client, but you still need to add an EDNS0 OPT record to the message you're sending https://godoc.org/github.com/miekg/dns#hdr-EDNS0 and https://godoc.org/github.com/miekg/dns#OPT.SetUDPSize

For retrying with TCP, you need to check the return if the msg has Truncated set and retry with dns.Client's Net set to "tcp"

**Comment by apparentlymart** [comment]
Retrying with the EDNS0 option and with TCP were implemented as part of #35.

**Comment by katbyte** [comment]
Hi @jaxxstorm ,

Just wanted to let you know that we have just released [v2.0.0](https://github.com/terraform-providers/terraform-provider-dns/blob/master/CHANGELOG.md#200-may-25-2018) of the dns provider containing a fix for this issue 🙂
